osu.Game/Online/Chat/Message.cs

```diff
@@ -42,17 +42,9 @@ public Message(long id)
             Id = id;
         }
 
-        public override bool Equals(object obj)
-        {
-            var objMessage = obj as Message;
-
-            return Id == objMessage?.Id;
-        }
+        public int CompareTo(Message other) => Id.CompareTo(other.Id);
 
-        public override int GetHashCode()
-        {
-            return Id.GetHashCode();
-        }
+        public bool Equals(Message other) => Id == other?.Id;

```

Comment:
OK, after a reading of referenced source, I found that `GetHashCode` is necessary for linq `Except`, as it uses an hash set. Two element with `Equals == true` won't be treated equal if they are in different buckets.
Now it should work.